subsys/net/conn_mgr/conn_mgr.c

```diff
@@ -0,0 +1,339 @@
+/*
+ * Copyright (c) 2019 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(conn_mgr, CONFIG_NET_CONNECTION_MANAGER_LOG_LEVEL);
+
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+#include <errno.h>
+#include <zephyr/net/net_core.h>
+#include <zephyr/net/net_if.h>
+#include <zephyr/net/net_mgmt.h>
+
+#include <zephyr/net/conn_mgr_connectivity.h>
+#include "conn_mgr_private.h"
+
+#if defined(CONFIG_NET_TC_THREAD_COOPERATIVE)
+#define THREAD_PRIORITY K_PRIO_COOP(CONFIG_NUM_COOP_PRIORITIES - 1)
+#else
+#define THREAD_PRIORITY K_PRIO_PREEMPT(7)
+#endif
+
+/* Internal state array tracking readiness, flags, and other state information for all available
+ * ifaces. Note that indexing starts at 0, whereas Zephyr iface indices start at 1.
```

Comment:
I am just wondering why the indexing cannot start from 1 for the iface_states too. In the getter/setter function you would then just do iface_states[index - 1]. Requires of course that one does not access the iface_states directly but via a function.

Reply:
IMO it could then can confusing in other place, where we simply iterate over the array - it be needed then not to use C-style array indexing (from 0) but start from 1 and use getter/setter. Perhaps it's just a matter of preference, but I don't think the current approach is that confusing (we no longer store the modified index in `last_iface_down/up`).

Reply:
Fair enough, if you think this is clear enough I can certainly live with this.